components/ILIAS/UI/src/Component/Progress/Instruction/Bar/Factory.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ */
+
+namespace ILIAS\UI\Component\Progress\Instruction\Bar;
+
+use ILIAS\UI\Component\Progress\Instruction\Instruction;
+
+/**
+ * @author Thibeau Fuhrer <thibeau@sr.solutions>
+ */
+interface Factory
+{
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     Factors a Progress Bar Instruction which orders the clientside Progress Bar
```

Comment:
```suggestion
     *     Creates a Progress Bar Instruction which orders the clientside Progress Bar
```

"to factor" means something else =)